lib/rules/no-deprecated-methods.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview Prevents usage of deprecated component lifecycle methods
+ */
+'use strict';
+
+const Components = require('../util/Components');
+const astUtil = require('../util/ast');
+const docsUrl = require('../util/docsUrl');
+
+function errorMessage(node, method) {
+  return `${node} should not use ${method}.`;
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevents usage of deprecated component lifecycle methods',
+      category: 'Possible Errors',
```

Comment:
`no-deprecated` is in 'Best Practices' category - maybe change it for consistency?